crates/bin/pcli/src/command/tx.rs

```diff
@@ -289,32 +263,50 @@ impl TxCmd {
                     text: memo.clone().unwrap_or_default(),
                 };
 
-                let plan = plan::send(
-                    app.fvk.account_group_id(),
-                    app.view.as_mut().context("view service initialized")?,
-                    OsRng,
-                    &values,
-                    fee,
-                    to,
-                    AddressIndex::new(*from),
-                    Some(memo_plaintext),
-                )
-                .await?;
+                let params = app
+                    .view
+                    .as_mut()
+                    .context("view service must be initialized")?
+                    .app_params()
+                    .await?;
```

Comment:
This is pulling the gas prices from the `AppParams` stored in the view service. How does the view service have the current `AppParams`?

I believe the `CompactBlock` currently contains a way for the full node to signal changes in parameters by including a new `AppParams` in the compact block. But this design was a compromise in the case where parameter changes were rare and infrequent, and would be much more problematic if the chain were to frequently adjust the gas prices.

In that case, we'd end up filling the compact blocks with the complete history of changes to gas prices, increasing bandwidth costs for all clients, and supplying them with lots of information they don't care about -- the intermediate history of gas prices is useless, only the gas prices at the moment they submit the application are relevant.

So, I wonder if we should be fetching the latest gas prices from the full node, rather than the view service, and not baking in an assumption that the view server has synchronized the current gas prices.

Alternatively, we could decide that the bandwidth cost of gas price changes isn't a big deal, and decide that the view service should synchronize gas prices.

Reply:
> I believe the CompactBlock currently contains a way for the full node to signal changes in parameters by including a new AppParams in the compact block.

It _did_ but this was broken (see #3107). I think you raise a valid issue in that updating the `AppParams` in the compact block _requires storing the entire `AppParams` structure in the compact block_. If only the gas prices are changing, and they are changing on a block-by-block basis, this is rather expensive.

Another potential alternative (that might be desirable when implementing #3107) is to have the param change updates in the compact block be at the component-level. That would mean we don't need to store a complete `AppParams` for a gas prices change. Or we poll the gas prices directly from the full node. 🤷

Reply:
What if:

1. We pull the gas prices out of the `AppParameters`, so that we can have a separation between "governance-controlled parameters that change infrequently and that clients can be updated on" and "auto-controlled prices that change frequently"
2. We add the gas prices into the `CompactBlock`, and decide that the chain will stream them to the client when they change.

This way we don't have to do a bigger architectural change about per-component parameter change updates etc. I think that the bandwidth costs might be pretty minimal, we could do some back of the envelope estimates.